src/main/java/com/example/finance/configuration/ClockProvider.java

```diff
@@ -0,0 +1,14 @@
+package com.example.finance.configuration;
+
+import org.springframework.stereotype.Component;
+
+import java.time.Clock;
+import java.time.Instant;
+
+@Component
+public class ClockProvider {
+
+    public Instant getDate() {
+        return Clock.systemUTC().instant();
```

Comment:
Try to not chain methods (except of streams or builders) :) Code will be easier to read